charts/rpc/templates/deployment.yaml

```diff
@@ -0,0 +1,169 @@
+{{- range $i := until (int .Values.replicaCount) }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "rpc.fullname"  $ }}
+  labels:
+    lavanet.xyz/node_moniker:  {{ include "rpc.moniker"  $ }}
+    lavanet.xyz/node_type: "rpc"
+    {{- include "rpc.labels" $ | nindent 4 }}
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      lavanet.xyz/node_moniker:  {{ include "rpc.moniker"  $ }}
+      lavanet.xyz/node_type: "rpc"
+      {{- include "rpc.selectorLabels" $ | nindent 6 }}
+  template:
+    metadata:
+      {{- with $.Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        lavanet.xyz/node_moniker:  {{ include "rpc.moniker"  $ }}
+        lavanet.xyz/node_type: "rpc"
+        {{- include "rpc.selectorLabels" $ | nindent 8 }}
+    spec:
+      {{- if $.Values.serviceAccountName }}
+      serviceAccountName: {{ $.Values.serviceAccountName }}
+      {{- end }}
+      securityContext:
+        {{- toYaml $.Values.podSecurityContext | nindent 8 }}
+      {{- if $.Values.snapshot.enabled }}
+      initContainers:
+        - name: download-and-install-snapshot
+          image: "{{ $.Values.snapshot.image.repository }}:{{ $.Values.snapshot.image.tag }}"
+          imagePullPolicy: {{ $.Values.snapshot.image.pullPolicy }}
+          command: {{ tpl ($.Values.snapshot.command | toYaml | nindent 12) . }}
+          volumeMounts:
+            - name: data-volume
+              mountPath: {{ $.Values.dataDir }}
```

Comment:
Not completely sure how making snapshots works now. Could you, please, give me a breakdown? 

What do you think about using `volumeSnapshot` objects? It should work in GCP as it works in AWS. We could have the following:

- A volume store snapshot on GCP that would serve as a "golden image" to be fed to the deployment.
- The volume could be created by (e.g.) a job that will run the client to sync up to the highest block. (We'd have to see how we could find the time to terminate the execution; Maybe by polling the RPC endpoint and checking whether the `syncing` endpoint is false?).
- Make a modification to the deployment's manifest to start up new pods via the latest snapshot (we could maybe pin it to the `latest` version somehow, and not bother with this). 

Maybe there's a better way of doing this. We should have those recurring jobs doing snapshots reasonably frequently.  We don't need to do any of this now, but it'd be good to find the best way to do it.

Reply:
Right now the only snapshot we have is one we are semi-manually creating with a github action in devops_repo: `/actions/workflows/snapshooter.yaml`

Once we get this running, using `volumeSnapshots` will be a much better Kubernetes native solution. And yes, a good way to do it is to figure out how quickly we want to spin up a new node and do snapshots based on that interval. 

For example, if we know it takes 5 mins to catch up 1000 blocks, and we want a new node to be spun up within 2 minutes we should do a snapshot every 200 blocks and delete old ones.

Right now the snapshot is a just tar of the blockchain data, stored on GCS. One nice thing about having the current snapshot-extracting method is external users could also easily use a snapshot if we decide to make it public. 

Do you know if we can do the same with `volumeSnapshots`?

Reply:
> Right now the only snapshot we have is one we are semi-manually creating with a github action in devops_repo: `/actions/workflows/snapshooter.yaml`
> 
> Once we get this running, using `volumeSnapshots` will be a much better Kubernetes native solution. And yes, a good way to do it is to figure out how quickly we want to spin up a new node and do snapshots based on that interval.
> 
> For example, if we know it takes 5 mins to catch up 1000 blocks, and we want a new node to be spun up within 2 minutes we should do a snapshot every 200 blocks and delete old ones.
> 
> Right now the snapshot is a just tar of the blockchain data, stored on GCS. One nice thing about having the current snapshot-extracting method is external users could also easily use a snapshot if we decide to make it public.
> 
Makes sense. Thanks!

> Do you know if we can do the same with `volumeSnapshots`?

I think we should approach it here a bit differently. We should definitely still support compressed community snapshots. The current mechanism seems more than good. For the Kubernetes workloads, we can directly snapshot the cloud volume store we're using. If I remember correctly, they're charging only for the delta between the two snapshots (i.e. the first one is the most expensive). 

In any case, we should decide what and how we want to open to the community, depending on what they actually would need and use. .